internal/storage/fs/snapshot.go

```diff
@@ -283,14 +289,15 @@ func (ss *storeSnapshot) addDoc(doc *ext.Document) error {
 
 		ns.flags[f.Key] = flag
 
+		var ruleRank int32 = 1
 		evalRules := []*storage.EvaluationRule{}
 		for _, r := range f.Rules {
```

Comment:
Take it or leave it: you could just use the loop variable as it is already incremented.
You would have to `+1` before you set it on the rank is all.
```suggestion
		evalRules := []*storage.EvaluationRule{}
		for i, r := range f.Rules {
```